modules/programs/waybar.nix

```diff
@@ -0,0 +1,388 @@
+      type = listOf waybarBarConfig;
+      default = [ ];
+      description = ''
+        Configuration for Waybar, see <link xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        [
+          {
+            layer = "top";
+            position = "top";
+            height = 30;
+            # Specify outputs to restrict to certain outputs, otherwise show on all outputs
+            output = [
+              "eDP-1"
+              "DP-1"
+            ];
+            modules-left = [ "sway/workspaces" "sway/mode" "wlr/taskbar" ];
+            modules-center = [ "sway/window" "custom/hello-from-waybar" ];
+            modules-right = [ "mpd" "custom/mymodule#with-css-id" "temperature" ];
+            modules = {
+              "sway/workspaces" = {
+                disable-scroll = true;
+                all-outputs = true;
+              };
+              "custom/hello-from-waybar" = {
+                format = "hello {}";
+                max-length = 40;
+                interval = "once";
+                # You may be interested in using symlinkJoin to merge all scripts in one derivation
+                # to have them all under one directory structure in the nix store
+                exec = "''${pkgs.writers.writeBashBin "hello-from-waybar" '''
+                  echo "from within waybar"
+                '''}/bin/hello-from-waybar";
+              };
+            };
+          }
+        ]
+      '';
+    };
+
+    systemd = {
+      enable = mkEnableOption "Waybar Systemd integration";
+
+      withSwayIntegration = mkOption {
+        type = bool;
+        default = false;
+        example = true;
+        description =
+          "Whether to bind the systemd service to the sway-session.target instead of graphical-session.target.";
+      };
+    };
+
+    style = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        CSS style of the bar.
+        See <link xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration/>"/> for the documentation.
+      '';
+      example = ''
+        * {
+          border: none;
+          border-radius: 0;
+          font-family: Source Code Pro;
+        }
+        window#waybar {
+          background: #16191C;
+          color: #AAB2BF;
+        }
+        #workspaces button {
+          padding: 0 5px;
+        }
+      '';
+    };
+  };
+
+  config = let
+    # Inspired by https://github.com/NixOS/nixpkgs/pull/89781
+    writePrettyJSON = name: x:
+      pkgs.runCommandNoCCLocal name { } ''
+        echo '${builtins.toJSON x}' | ${pkgs.jq}/bin/jq . > $out
+      '';
+
+    configSource = let
+      # Removes nulls because Waybar ignores them for most values
+      removeNulls = filterAttrs (_: v: !isNull v);
+      # Makes the actual valid configuration Waybar accepts (strips our custom settings before converting to JSON)
```

Comment:
Done. The longest line is now 112 characters long (a link).
I've further split documentation string since white-space is not interpreted in XML tags.